tests/contrib/io/dask/test_dask_parquet.py

```diff
@@ -0,0 +1,290 @@
+            ParquetDaskDataSet(
+                filepath=FILENAME, credentials=bad_credentials
+            )
+
+    def test_incorrect_credentials_load(self):
+        """Test that incorrect credential keys won't instantiate dataset."""
+        pattern = "unexpected keyword argument"
+        with pytest.raises(TypeError, match=pattern):
+            ParquetDaskDataSet(
+                filepath=FILENAME,
+                credentials={"access_token": "TOKEN", "access_key": "KEY"},
+            )
+
+    @pytest.mark.parametrize(
+        "bad_credentials",
+        [{"aws_access_key_id": None, "aws_secret_access_key": None}, {}, None],
+    )
+    def test_empty_credentials_load(self, bad_credentials):
+        parquet_data_set = ParquetDaskDataSet(
+            filepath=FILENAME, credentials=bad_credentials
+        )
+        pattern = r"Failed while loading data from data set ParquetS3DaskDataSet\(.+\)"
+        with pytest.raises(DataSetError, match=pattern):
+            parquet_data_set.load()
+
+    def test_pass_credentials(self, mocker):
+        """Test that AWS credentials are passed successfully into boto3
+        client instantiation on creating S3 connection."""
+        mocker.patch("s3fs.core.boto3.Session.client")
+        s3_data_set = ParquetDaskDataSet(
+            filepath=FILENAME, credentials=AWS_CREDENTIALS
+        )
+        pattern = r"Failed while loading data from data set ParquetS3DaskDataSet\(.+\)"
+        with pytest.raises(DataSetError, match=pattern):
+            s3_data_set.load()
+
+        assert s3fs.core.boto3.Session.client.call_count == 1
+        args, kwargs = s3fs.core.boto3.Session.client.call_args_list[0]
+        assert args == ("s3",)
+        for k, v in AWS_CREDENTIALS.items():
+            assert kwargs[k] == v
+
+    @pytest.mark.usefixtures("mocked_s3_object")
+    def test_load_data(self, s3_data_set, dummy_dataframe):
+        """Test loading the data from S3."""
+        loaded_data = s3_data_set.load()
+        assert_frame_equal(loaded_data, dummy_dataframe)
+
+    @pytest.mark.usefixtures("mocked_s3_object")
+    def test_save_data(self, s3_data_set):
+        """Test saving the data to S3."""
+        new_data = pd.DataFrame(
+            {"col1": ["a", "b"], "col2": ["c", "d"], "col3": ["e", "f"]}
+        )
+        new_dask_data = dd.from_pandas(new_data, npartitions=2)
+        s3_data_set.save(new_dask_data)
+        loaded_data = s3_data_set.load()
+        assert_frame_equal(loaded_data, new_dask_data)
+
+    @pytest.mark.usefixtures("mocked_s3_bucket")
+    def test_exists(self, s3_data_set, dummy_dataframe):
+        """Test `exists` method invocation for both existing and
+        nonexistent data set."""
+        assert not s3_data_set.exists()
+        s3_data_set.save(dummy_dataframe)
+        assert s3_data_set.exists()
+
+
+@pytest.mark.usefixtures("mocked_s3_bucket", "s3fs_cleanup")
+class TestParquetS3DaskDataSetVersioned:
```

Comment:
As mentioned above, this dataset implementation does not support versioning, so this test class is redundant (and, probably, doesn't pass)